tests/unit/repos/test_factory.py

```diff
@@ -0,0 +1,53 @@
+from unittest.mock import Mock
+
+import pytest
+from yarl import URL
+
+from eligibility_signposting_api.repos.factory import dynamodb_resource_factory, s3_service_factory
+
+
+@pytest.fixture
+def mock_session():
+    return Mock()
+
+
+def test_dynamodb_resource_factory_with_endpoint(mock_session):
+    mock_resource = Mock()
```

Comment:
```suggestion
    mock_resource = MagicMock(spec=ServiceResource)
```
You'll also need `from boto3.resources.base import ServiceResource`.